app/controllers/UrlController.php

```diff
@@ -9,188 +9,200 @@
+                        'matchCallback' => function () {
+                            return !Yii::$app->user->isGuest ? Yii::$app->user->can('urlCreate') : true;
+                        }
+                    ],
+                    [
+                        'allow' => true,
+                        'actions' => ['delete'],
+                        'verbs' => ['POST'],
+                        'roles' => ['urlManage'],
+                    ],
+                    [
+                        'allow' => true,
+                        'actions' => ['index', 'update'],
+                        'roles' => ['urlManage'],
+                    ],
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * Lists all NixShortUrls models.
+     * @return string
+     */
+    public function actionIndex(): string
+    {
+        $searchModel = new NixShortUrlsSearch();
+        $dataProvider = $searchModel->search(Yii::$app->request->getQueryParams());
+
+        return $this->render(
+            'index',
+            [
+                'dataProvider' => $dataProvider,
+                'searchModel' => $searchModel,
+            ]
+        );
+    }
+
+    /**
+     * Updates an existing NixShortUrls model.
+     * If update is successful, the browser will be redirected to the 'view' page.
+     *
+     * @param integer $id
+     *
+     * @return string
+     */
+    public function actionUpdate($id): string
+    {
+        $model = $this->findModel($id);
+
+        if ($model->load(Yii::$app->request->post()) && $model->save()) {
+            return $this->redirect(['index']);
+        }
+
+        return $this->render(
+            'update',
+            [
+                'model' => $model,
+            ]
+        );
     }
 
-    return $this->render('update', [
-        'model' => $model,
-    ]);
-  }
-
-  /**
-   * Deletes an existing NixShortUrls model.
-   * If deletion is successful, the browser will be redirected to the 'index' page.
-   * @param integer $id
-   * @return mixed
-   * @throws NotFoundHttpException
-   * @throws \yii\db\StaleObjectException
-   */
-  public function actionDelete($id)
-  {
-    $this->findModel($id)->delete();
-
-    return $this->redirect(['index']);
-  }
-
-  /**
-   * Finds the NixShortUrls model based on its primary key value.
-   * If the model is not found, a 404 HTTP exception will be thrown.
-   * @param integer $id
-   * @return NixShortUrls the loaded model
-   * @throws NotFoundHttpException if the model cannot be found
-   */
-  protected function findModel($id)
-  {
-    if (($model = NixShortUrls::findOne($id)) === null) {
-        throw new NotFoundHttpException('The requested page does not exist.');
+    /**
+     * Deletes an existing NixShortUrls model.
+     * If deletion is successful, the browser will be redirected to the 'index' page.
+     *
+     * @param integer $id
+     *
+     * @return Response
+     */
+    public function actionDelete($id): Response
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid variables with short names like $id. Configured minimum length is 3.](https://app.codacy.com/app/gaalferov/yii2-minify-url/pullRequest?prid=4708268)